gnomad_hail/utils/sample_qc.py

```diff
@@ -278,86 +278,55 @@ def assign_platform_from_pcs(
-        max_y_female_call_rate: float = 0.15,
-        min_y_male_call_rate: float = 0.8
+        mt: hl.MatrixTable,
+        ploidy_ht: hl.Table,
+        f_stat_cutoff: float,
+        sites_ht: Optional[hl.Table],
+        aaf_expr: Optional[str] = None,
+        gt_expr: str = 'GT',
+        aaf_threshold: float = 0.001,
+        male_threshold: float = 0.75,
+        female_threshold: float = 0.5,
 ) -> hl.Table:
     """
     Imputes sample sex based on X-chromosome heterozygosity and Y-callrate (if Y calls are present)
 
-    :param x_mt:
-    :param y_mt:
-    :param platform_ht:
-    :param male_min_f_stat:
-    :param female_max_f_stat:
-    :param min_male_y_sites_called:
-    :param max_y_female_call_rate:
-    :param min_y_male_call_rate:
+    :param mt: Input MatrixTable
+    :param ploidy_ht: Table with imputed sex chromosome ploidies
+    :param f_stat_cutoff: f-stat to roughly divide 'XX' from 'XY' samples. Assumes XX samples are below cutoff and XY are above cutoff.
+    :param sites_ht: Optional Table to use. If present, filters input MatrixTable to sites in this Table prior to imputing sex.
+    :param aaf_expr: Optional. Name of field in input MatrixTable with alternate allele frequency.
+    :param gt_expr: Name of entry field storing the genotype. Default: 'GT'
+    :param float aaf_threshold: Minimum alternate allele frequency to be used in f-stat calculations.
+    :param float male_threshold: Threshold above which a sample will be called male.
+    :param float female_threshold: Threshold below which a sample will be called female
     :return:
     """
-    logger.info("Imputing samples sex")
-
-    x_mt = hl.filter_intervals(x_mt, [hl.parse_locus_interval(x_contig) for x_contig in get_reference_genome(x_mt.locus).x_contigs])
-    x_ht = hl.impute_sex(x_mt.GT, aaf_threshold=0.05, female_threshold=female_max_f_stat, male_threshold=male_min_f_stat)
-    y_mt = hl.filter_intervals(y_mt, [hl.parse_locus_interval(y_contig) for y_contig in get_reference_genome(y_mt.locus).y_contigs])
-    y_mt = y_mt.filter_rows(y_mt.locus.in_y_nonpar())
-    sex_ht = y_mt.annotate_cols(
-        qc_platform=platform_ht[y_mt.col_key].qc_platform,
-        is_female=x_ht[y_mt.col_key].is_female,
-        y_call_rate=hl.agg.fraction(hl.is_defined(y_mt.GT)),
-        n_y_sites_called=hl.agg.count_where(hl.is_defined(y_mt.GT)),
-        **{f'x_{ann}': x_ht[y_mt.col_key][ann] for ann in x_ht.row_value if ann != 'is_female'}
-    ).cols()
-
-    mean_male_y_sites_called = sex_ht.aggregate(hl.agg.filter(~sex_ht.is_female, hl.agg.group_by(sex_ht.qc_platform, hl.agg.mean(sex_ht.n_y_sites_called))))
-    y_call_rate_stats = sex_ht.aggregate(
-        hl.agg.filter(
-            hl.is_defined(sex_ht.is_female),
-            hl.agg.group_by(hl.tuple([sex_ht.qc_platform, sex_ht.is_female]),
-                            hl.agg.stats(sex_ht.y_call_rate)
-                            )
-        )
-    )
+    logger.info("Filtering mt to chrX")
```

Comment:
Maybe we should get the list X-contigs before and print the actual list in the INFO ?

Reply:
Otherwise maybe use "x contig(s)" instead?